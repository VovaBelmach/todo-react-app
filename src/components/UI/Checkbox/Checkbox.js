import React from "react";
import PropTypes from "prop-types";
import styles from "./Checkbox.module.css";

const Checkbox = (props) => {
  return (
    <>
      <label>
        <input
          type="checkbox"
          className={styles.checkbox}
          id={props.id}
          name={props.id}
          checked={props.isCompleted}
          onChange={() => {}}
          onClick={props.onClick}
          disabled={props.isDisabled}
        />
        {props.description}
      </label>
    </>
  );
};

Checkbox.propTypes = {
  id: PropTypes.any,
  description: PropTypes.any,
  isCompleted: PropTypes.bool,
  isDisabled: PropTypes.bool,
  onClick: PropTypes.func,
};

export default Checkbox;
