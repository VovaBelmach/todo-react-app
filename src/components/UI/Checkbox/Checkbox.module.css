label {
    display: flex;
    align-items: center;
}

.checkbox {
    appearance: none;
    border: 2px solid #ccc;
    border-radius: 50%;
    width: 20px;
    height: 20px;
    margin-right: 10px;
    cursor: pointer;
    position: relative;
}

.checkbox:checked::before {
    content: "";
    display: block;
    position: absolute;
    inset: 2px;
    background-color: #0080ff;
    border-radius: 50%;
}

.checkbox:disabled {
    background-color: #a1a1a1;
    cursor: not-allowed;
}