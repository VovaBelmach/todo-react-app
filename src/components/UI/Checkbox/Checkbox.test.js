import React from "react";
import renderer from "react-test-renderer";
import Checkbox from "./Checkbox";

describe("Checkbox", () => {
    it("should renders correctly", () => {
        const tree = renderer.create(<Checkbox />).toJSON();
        expect(tree).toMatchSnapshot();
    })
})