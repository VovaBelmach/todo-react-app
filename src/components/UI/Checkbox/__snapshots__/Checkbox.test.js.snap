// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox should renders correctly 1`] = `
<label>
  <input
    className="checkbox"
    onChange={[Function]}
    type="checkbox"
  />
</label>
`;
